### DISCUSSION TITLE
Calling .js  using Microsoftclearscript works for Desktop project, but not Browser.

### DISCUSSION THREAD

**Discussion by Coz540** [AvaloniaUI/Avalonia#15868] [Q&A]
Hi, I'm using Microsoft clearscript complete 7.4.5 to execute some .js files, which works perfectly when running as Desktop application. trying to run as a browser application Wont compile becuase of this error.. : 

Error		Parameter types of pinvoke callback method 'System.Void <get_GetHostObjectNamedPropertyFastMethodPtr>g__Thunk|110_0(System.IntPtr, Microsoft.ClearScript.V8.SplitProxy.StdString+Ptr, Microsoft.ClearScript.V8.SplitProxy.V8Value+Ptr)' needs to be blittable.	xplatMapTest.Browser	C:\Program Files\dotnet\packs\Microsoft.NET.Runtime.WebAssembly.Sdk\8.0.2\Sdk\WasmApp.Native.targets	338		

Im guessing its not liking an included Function inside clearscript...? 

I know next to nothing about WASM. however, since the browser app seems to run javascript itself... should I remove the clearscript dependency? but How do I do this conditionally so leave it in for Desktop but conditionally call the browser app code to evaluate the JS. ? or is there a better way to run the Javascript code for all avalonia project types. without creating a clearscript Engine to evaluate the .JS files ?

**Comment by maxkatz6** [comment]
> Im guessing its not liking an included Function inside clearscript...?

ClearScript uses native assembly to execute java script code. And it seems to only be implemented on desktop platforms https://github.com/microsoft/ClearScript?tab=readme-ov-file#v8-native-assembly-packages.

So, no, you can't use ClearScript there.
Related https://github.com/microsoft/ClearScript/issues/410

**Reply by maxkatz6** [reply]
Alternative could be Jint compiler - https://github.com/sebastienros/jint
It's fully managed and doesn't require any native assemblies to execute java script code.
Note, I never tried Jint with Browser/WASM personally.

Also, you might consider using real browser javascript code via [JSImport/JSExport](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0) interop. It will only work on Browser though, so you would need to abstract scripting between different platforms.
